app/src/main/java/ch/epfl/culturequest/database/FireDatabase.java

```diff
@@ -194,4 +214,148 @@ public CompletableFuture<List<Profile>> getTopNProfiles(int n) {
         return future;
     }
 
+    @Override
+    public CompletableFuture<AtomicBoolean> uploadPost(Post post) {
+        CompletableFuture<AtomicBoolean> future = new CompletableFuture<>();
+        DatabaseReference usersRef = database.getReference("posts").child(post.getUid()).child(String.valueOf(post.getPostid()));
+        usersRef.setValue(post).addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                future.complete(new AtomicBoolean(true));
+            } else {
+                future.complete(new AtomicBoolean(false));
+            }
+        });
+        return future;
+    }
+
+    @Override
+    public CompletableFuture<AtomicBoolean> removePost(Post post) {
+        CompletableFuture<AtomicBoolean> future = new CompletableFuture<>();
+        DatabaseReference usersRef = database.getReference("posts").child(post.getUid()).child(post.getPostid());
+        usersRef.removeValue((error, ref1) -> {
+            if (error == null) {
+                future.complete(new AtomicBoolean(true));
+            } else {
+                future.complete(new AtomicBoolean(false));
+            }
+        });
+        return future;
+    }
+
+    @Override
+    public CompletableFuture<List<Post>> getPosts(String UId, int limit, int offset) {
```

Comment:
i think a method to retrieve all the posts of a user would be useful, to display in the profile and when looking at other ppls profiles bc for now there is a limit but we don't know how many posts a user has. i don't know if i just didn't see it

Reply:
I think that like Hugo did with getting the top n profile for the leaderborad we should have a paginated view, because it will be impossible to fetch too much posts at the same time. The recycler view is made for such pager layout and fetch the rest.

Reply:
Ahhh i see, makes sense